src/Data/Semigroup/Traversable.hs

```diff
@@ -29,11 +31,19 @@ import Data.Semigroup
 #endif
 import Data.Semigroup.Traversable.Class
 import Data.Functor.Bind.Class
+import GHC.Generics (Rep1, Generic1, to1, from1)
 
 -- | Default implementation of 'foldMap1' given an implementation of 'Traversable1'.
 foldMap1Default :: (Traversable1 f, Semigroup m) => (a -> m) -> f a -> m
 foldMap1Default f = getConst . traverse1 (Const . f)
 
+gtraverse1 ::
```

Comment:
Two potential caveats, which I think are worth mentioning in the Haddocks, is that `gtraverse1` won't work in the following cases:

* When you have a nullary data constructor (e.g., `data Foo a = MkFoo`). This is pretty sensible, since a nullary data constructor has no field of type `a`, but it is worth this pointing out.
* When you have a data constructor with a field whose type doesn't mention `a` (e.g., `data Bar a = MkBar a Int`). This limitation is a bit more surprising, as one _could_ define a `Traversable1` instance for `Bar` by hand, but the `Generic1`-based approach used here isn't smart enough to realize this.